src/lib/fetch.d.ts

```diff
@@ -0,0 +1,129 @@
+/// The fetch API standard can be found at https://fetch.spec.whatwg.org/
+/// Definitions by: Ryan Graham <https://github.com/ryan-codingintrigue>, Kagami Sascha Rosylight <https://github.com/saschanaz>, Robin Van den Broeck <https://github.com/gamesmaxed>
+
+/// <reference path="lib.streams.d.ts" />
+
+interface Window {
+    fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
+}
+declare var fetch: typeof window.fetch;
+
+declare type HeadersInit = Headers | string[][] | { [key: string]: string };

```

Comment:
Maybe `([string, string])[]` instead of `string[][]`?

I also checked and was slightly disappointed to see it cannot accept arbitrary `Iterable<[string, string]>` :/